### PR TITLE
analog: add -lm flag for Linux and set os specific build targets

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
This formula still has a Mavericks (!) bottle, so let's merge this without bottles as there's no need to discard the old ones.

**Comment by carlocab** [comment]
That's weird:

```
==> make CC=clang CFLAGS= DEFS='-DLANGDIR="/opt/homebrew/Cellar/analog/6.0_1/share/analog/lang/"' -DHAVE_ZLIB LIBS=-lz -lm OSX
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- D
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- H
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- A
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- V
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- E
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- _
/Applications/Xcode.app/Contents/Developer/usr/bin/make: invalid option -- Z
```

File: Formula/analog.rb [hunk 1]
```diff
@@ -26,11 +26,19 @@ class Analog < Formula
...
-                   "LIBS=-lz",
-                   "OS=OSX"
+    args = %W[
+      CC=#{ENV.cc}
+      CFLAGS=#{ENV.cflags}
+      DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"' -DHAVE_ZLIB
+      LIBS=-lz -lm
+    ]
+    on_macos do
+      args << "OSX"
```
Comment (carlocab) [inline Formula/analog.rb:36]:
```suggestion
      args << "OS=OSX"
```

File: Formula/analog.rb [hunk 2]
```diff
@@ -26,11 +26,19 @@ class Analog < Formula
...
+    end
+    on_linux do
+      args << "UNIX"
```
Comment (carlocab) [inline Formula/analog.rb:39]:
```suggestion
      args << "OS=UNIX"
```
?

**Comment by iMichka** [comment]
Yes it's me, I broke the syntax. It should be fixed now.

File: Formula/analog.rb [hunk 3]
```diff
@@ -26,11 +26,19 @@ class Analog < Formula
...
+      DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"' -DHAVE_ZLIB
+      LIBS=-lz -lm
+    ]
+    on_macos do
+      args << "OS=OSX"
+    end
+    on_linux do
+      args << "OS=UNIX"
+    end
+    system "make", args
```
Comment (Bo98) [inline Formula/analog.rb:41]:
```suggestion
    system "make", *args
```

File: Formula/analog.rb [hunk 4]
```diff
@@ -26,11 +26,19 @@ class Analog < Formula
...
   def install
-    system "make", "CC=#{ENV.cc}",
-                   "CFLAGS=#{ENV.cflags}",
-                   "DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"' -DHAVE_ZLIB",
-                   "LIBS=-lz",
-                   "OS=OSX"
+    args = %W[
+      CC=#{ENV.cc}
+      CFLAGS=#{ENV.cflags}
+      DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"' -DHAVE_ZLIB
```
Comment (Bo98) [inline Formula/analog.rb:32]: This passes it as two args (due to the space, as this is a space-sparated array), when it needs to be one.

File: Formula/analog.rb [hunk 5]
```diff
@@ -26,11 +26,19 @@ class Analog < Formula
...
   def install
-    system "make", "CC=#{ENV.cc}",
-                   "CFLAGS=#{ENV.cflags}",
-                   "DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"' -DHAVE_ZLIB",
-                   "LIBS=-lz",
-                   "OS=OSX"
+    args = %W[
+      CC=#{ENV.cc}
+      CFLAGS=#{ENV.cflags}
+      DEFS='-DLANGDIR=\"#{pkgshare}/lang/\"'\ -DHAVE_ZLIB
```
Comment (carlocab) [inline Formula/analog.rb:32]: I think you don't need to escape double-quotes here, actually.

Comment (iMichka) [reply Formula/analog.rb]: Right. That was legacy code. I had to escape the space though.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**